src/pages/Auth/Auth.tsx

```diff
@@ -1,95 +1,110 @@
-        `https://www.googleapis.com/oauth2/v1/userinfo?access_token=${userToken}`,
-        {
-          headers: {
-            Authorization: `Bearer ${userToken}`,
-            Accept: 'application/json',
-          },
-        },
-      );
-
-      const { name, email, picture }: UserInfo = data;
-
-      dispatch(setUserInfo({ name, email, picture }));
-    } catch (e) {
-      console.log(e);
+      if (tokenResponse) {
+        const { name, email, picture }: UserInfo = jwtDecode(
+          tokenResponse.credential || '',
+        );
+
+        dispatch(setUserInfo({ name, email, picture }));
+
+        localStorage.setItem(USER_TOKEN_KEY, tokenResponse.credential || '');
+        dispatch(setUserToken(tokenResponse.credential || ''));
+      } else {
+        const credential = localStorage.getItem(USER_TOKEN_KEY);
+        const { name, email, picture }: UserInfo = jwtDecode(credential || '');
+
+        dispatch(setUserInfo({ name, email, picture }));
+
+        localStorage.setItem(USER_TOKEN_KEY, credential || '');
+        dispatch(setUserToken(credential || ''));
+      }
+    } catch (err) {
+      console.log('Login failed: ', err);
     }
   };
 
-  useEffect(() => {
-    const localToken = localStorage.getItem(USER_TOKEN_KEY);
+  const logOutHandle = () => {
+    googleLogout();
+    localStorage.removeItem(USER_TOKEN_KEY);
+    dispatch(logout());
+  };
 
-    if (localToken) {
-      dispatch(setUserToken(localToken));
-    }
-  }, []);
+  // const getUserInfo = async () => {
+  //   try {
+  //     const { data } = await axios.get(
+  //       `https://www.googleapis.com/oauth2/v1/userinfo?access_token=${userToken}`,
+  //       {
+  //         headers: {
+  //           Authorization: `Bearer ${userToken}`,
+  //           Accept: 'application/json',
+  //         },
+  //       },
+  //     );
+
+  //     const { name, email, picture }: UserInfo = data;
+
+  //     dispatch(setUserInfo({ name, email, picture }));
+  //   } catch (e) {
+  //     console.log(e);
+  //   }
+  // };
+
+  // useEffect(() => {
+  //   if (userToken) {
+  //     getUserInfo();
+  //   }
+  // }, [userToken]);
 
   useEffect(() => {
-    if (userToken) {
-      getUserInfo();
-    }
-  }, [userToken]);
+    googleAuth(null);
+  }, []);
 
   return (
     <div className={styles.auth}>
-      <h2>React Google Login</h2>
-      {name ? (
+      {userToken ? (
         <div>
           <img src={picture} alt="user image" />
           <h3>User Logged in</h3>
           <p>Name: {name}</p>
           <p>Email Address: {email}</p>
           <br />
           <br />
-          <button onClick={logOut}>Log out</button>
+          <button onClick={logOutHandle}>Log out</button>
         </div>
       ) : (
-        <button onClick={() => login()}>Sign in with Google 🚀 </button>
+        <GoogleLogin
+          logo_alignment="center"
+          text="signin"
+          theme="outline"
+          width="350px"
+          onSuccess={googleAuth}
```

Comment:
Maybe need to change this, change only if ts error
```suggestion
          onSuccess={(res) => googleAuth(res)}
```